src/main/java/sh/okx/railswitch/listener/DetectorRailActivateListener.java

```diff
@@ -22,71 +22,95 @@
 import java.util.Optional;
 
 public class DetectorRailActivateListener implements Listener {
+
+  private static final String NORMAL_SWITCH_TYPE = "[destination]";
+  private static final String INVERTED_SWITCH_TYPE = "[!destination]";
+
   private final RailSwitch plugin;
+  private long timingsStartTime = System.currentTimeMillis();
 
   public DetectorRailActivateListener(RailSwitch plugin) {
     this.plugin = plugin;
   }
 
+  private void concludeTiming() {
+    if (plugin.isTimings()) {
+      plugin.getLogger().info("Took " + (System.currentTimeMillis() - timingsStartTime) + "ms");
+    }
+  }
+
   @EventHandler
-  public void on(BlockRedstoneEvent e) {
-    Block block = e.getBlock();
-    // check if block is a detector rail
-    if (block == null || block.getType() != Material.DETECTOR_RAIL || e.getNewCurrent() != 15) {
+  public void onRailSwitch(BlockRedstoneEvent event) {
+    // If Citadel is not enabled, do nothing
+    if (!Bukkit.getPluginManager().isPluginEnabled("Citadel")) {
```

Comment:
RailSwitch should work without citadel, it does not fundamentally depend on citadel

Reply:
Reverted